dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -230,6 +236,32 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
         return tool;
     }
 
+    private static List<ImageData> processImageDataForWorkflowVersions(final Version version, final ToolVersion toolVersion) {
+        toolVersion.setIsProduction(true);
+        Set<Image> images = version.getImages();
+        List<ImageData> trsImages = new ArrayList<>();
+
+        for (Image image : images) {
+            ImageData imageData = new ImageData();
+            imageData.setImageType(ImageType.DOCKER);
+            //TODO: For workflows, it's just quay for now, but that'll change. Should add column in image table to store registry.
+            imageData.setRegistryHost("quay.io");
+            imageData.setImageName(constructName(Arrays.asList(image.getRepository(), image.getTag())));
+            List<Checksum> trsChecksums = new ArrayList<>();
+            List<io.dockstore.webservice.core.Checksum> checksumList = image.getChecksums();
+
+            for (io.dockstore.webservice.core.Checksum checksum : checksumList) {
```

Comment:
This is fine, but in the future you might want to look at using `map` instead as an alternative and slightly more functional way (fix not required). And also for the outer loop.